### PR TITLE
Add README section about whether PinkyPromise is best for the reader.

### PR DESCRIPTION
**Pull request by kconner**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,14 +4,29 @@ A tiny Promises library.
...
 
-- `Result` - A value or error. `Result` adapts the return-or-throw function pattern for asynchronous operations with callbacks.
+- `Result` - A value or error. `Result` adapts the return-or-throw function pattern for use with asynchronous callbacks.
 - `Promise` - An operation that produces a `Result` sometime after it is called. `Promise`s can be composed and sequenced.
 
-Please see `PinkyPromise.playground` for examples and insight.
+## Should I use this?
 
-PinkyPromise is meant to be a lightweight functional tool that does a lot of heavy lifting. A natural next step beyond these two types is an [Observable](https://www.youtube.com/watch?v=looJcaeboBY). You might use PinkyPromise as a stepping stone on the way to learning [RxSwift](https://github.com/ReactiveX/RxSwift), which we recommend.
+There are lots of promise libraries. PinkyPromise:
```
Comment (cschandler) [inline README.md]: Odd beginning sentence that reads like just a statement.  I would either cut it, or state a little more explicitly where this library differs.

File: README.md [hunk 2]
```diff
@@ -4,14 +4,29 @@ A tiny Promises library.
...
+
+- Is lightweight
+- Is tested
+- Embraces the Swift language with airtight type system contracts and `throw` / `catch`
+- Embraces functional style with immutable values and value transformations
+- Is a great way for Objective-C programmers to learn functional style in Swift
+- Is easy to extend with your own Promise transformations
+
+PinkyPromise is meant to be a lightweight functional tool that's easy to learn but does a lot of heavy lifting. Fuller-featured implementations include [Result](https://github.com/antitypical/Result) and [PromiseKit](http://promisekit.org).
```
Comment (cschandler) [inline README.md]: `lightweight functional tool that's easy to learn but does a lot of heavy lifting` almost the exact same sentence as in the previous section.  Maybe that was intentional, but it reads awkwardly if you're actually going through the whole thing.

Comment (cschandler) [inline README.md]: `Fuller-featured` More fully featured

File: README.md (hunk 1 above)
Comment (cschandler) [inline README.md]: `lightweight functional tool` functional as in the mathematical sense, or the utility sense?  Maybe it's already clear enough given the context, but it made me pause to think about it.

**Comment by eriklamanna** [comment]
So overall comment, I'm not sure I would get the full understanding of the reason I would want to use this as a newbie. I have found in the past, where it helps in code to maybe show the old way vs. the new way that you would do something with the library.

**Comment by kconner** [comment]
That's a good idea. The playground goes into detail, but we could use an example section up front in the readme.

**Pull request merged**